### PR TITLE
[Merged by Bors] - chore(RingTheory): split Finiteness.lean into many smaller files

### PR DESCRIPTION
**Pull request by Vierkantor**
This PR splits off `RingTheory/Finiteness.lean`, where finitely generated (sub)modules are defined, into many different files:

 * `RingTheory/Finiteness/Basic.lean`: basic properties and instances of `Submodule.FG` and `Module.Finite`
 * `RingTheory/Finiteness/Bilinear.lean`: results involving bilinear maps
 * `RingTheory/Finiteness/Cardinality.lean`: finite modules and types with finite cardinality
 * `RingTheory/Finiteness/Defs.lean`: definition of finitely generated (sub)modules and ideals
 * `RingTheory/Finiteness/Ideal.lean`: results on finitely generated ideals
 * `RingTheory/Finiteness/Lattice.lean`: finiteness of suprema in the submodule lattice
 * `RingTheory/Finiteness/Nakayama.lean`: Nakayama's lemma
 * `RingTheory/Finiteness/Nilpotent.lean`: results on nilpotent maps on finite modules
 * `RingTheory/Finiteness/Prod.lean`: results involving products of (sub)modules
 * `RingTheory/Finiteness/Projectve.lean`: results on finite projective modules
 * `RingTheory/Finiteness/Finsupp.lean`: results involving (`D`)`Finsupp`
 * `RingTheory/Finiteness/Subalgebra.lean`: results involving (sub)algebras
 * `RingTheory/Finiteness/TensorProduct.lean`: finiteness of the tensor products

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Finiteness/TensorProduct.lean [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import Mathlib.Algebra.Algebra.Bilinear
+import Mathlib.Algebra.Module.Submodule.Pointwise
+import Mathlib.LinearAlgebra.Isomorphisms
+import Mathlib.RingTheory.Finiteness.Basic
+import Mathlib.RingTheory.Finiteness.Bilinear
+import Mathlib.RingTheory.Ideal.Quotient.Basic
+
+/-!
+# Finiteness of the tensor product of (sub)modules
+
```
Comment (Vierkantor) [inline Mathlib/RingTheory/Finiteness/TensorProduct.lean:15]:
```suggestion

TODO: this file can probably be merged with `RingTheory.TensorProduct.Finite`.
```

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by Vierkantor** [comment]
Looks like Bors failing is the fault of this one so:

bors r-

**Comment by Vierkantor** [comment]
This doesn't seem to have conflicts with the current `master`, so I will wait for bors to merge the conflicting PR and hope we don't lose the race again :)

**Pull request closed without merging**